userspace/libsinsp/sinsp_filter_transformer.cpp

```diff
@@ -180,6 +201,50 @@ bool sinsp_filter_transformer::transform_values(std::vector<extract_value_t>& ve
 			return true;
 		});
 	}
+	case FTR_LEN: {
+		assert((void("len() type must be PT_UINT64"), t == PT_UINT64));
+		if(is_list) {
+			uint64_t len = static_cast<uint64_t>(vec.size());
+			auto stored_val = store_scalar(len);
+			vec.clear();
+			vec.push_back(stored_val);
+			return true;
+		}
+
+		// not a list: could be string or buffer
+		bool is_string = false;
+		switch(original_type) {
+		case PT_CHARBUF:
+		case PT_FSPATH:
+		case PT_FSRELPATH:
+			is_string = true;
+			break;
+		case PT_BYTEBUF:
+			is_string = false;
+			break;
+		default:
+			return false;
+		}
+
+		for(std::size_t i = 0; i < vec.size(); i++) {
```

Comment:
Question -- is there really a scenario where the field is not a list-type and `vec.size() > 1`? Should we put an assert?

Reply:
I honestly do not know, but tests were using it like this.

Reply:
If you can't think of a corner case (such as plugins) I'll remove the loop as it simplifies things.

Reply:
done